storage/src/adb/any.rs

```diff
@@ -498,34 +492,38 @@ impl<E: RStorage + Clock + Metrics, K: Array, V: Array, H: CHasher, T: Translato
     }
 
     /// Return true if the given sequence of `ops` were applied starting at location `start_loc` in
-    /// the log with the provided root hash.
-    pub fn verify_proof(
+    /// the log with the provided root.
+    pub async fn verify_proof(
         hasher: &mut H,
         proof: &Proof<H>,
         start_loc: u64,
         ops: &[Operation<K, V>],
-        root_hash: &H::Digest,
-    ) -> bool {
+        root_digest: &H::Digest,
+    ) -> Result<bool, Error> {
         let start_pos = leaf_num_to_pos(start_loc);
         let end_loc = start_loc + ops.len() as u64 - 1;
         let end_pos = leaf_num_to_pos(end_loc);
 
+        let mut hasher = Basic::new(hasher);
         let digests = ops
             .iter()
-            .map(|op| Any::<E, _, _, _, T>::op_digest(hasher, op))
+            .map(|op| Any::<E, _, _, _, T>::op_digest(&mut hasher, op))
             .collect::<Vec<_>>();
 
-        proof.verify_range_inclusion(hasher, digests, start_pos, end_pos, root_hash)
+        let r = proof
```

Comment:
nit: I think you can just return `proof...await` here?

Reply:
Almost, needed to map the error (done)